### ISSUE TITLE
SpaProxy doesn't set headers such that it would work with all potential SPAs

### ISSUE THREAD

**Issue by keithn** [dotnet/aspnetcore#33984]
in the package https://www.nuget.org/packages/microsoft.aspnetcore.spaproxy

the code that checks for whether spa is running

```
var response = await httpClient.GetAsync(_options.ServerUrl, cancellationTokenSource.Token);
                var running = response.IsSuccessStatusCode;
```
will 404 on things like Vue 3 running with Vite because the header "Accept" hasn't been set.    This will result in the system thinking it isn't running and will launch a duplicate instance.   I think by default the following should be set

```
httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("*/*"))
```

**Comment by johanndev** [comment]
Already addressed - see #33115.

**Comment by javiercn** [comment]
@keithn thanks for contacting us.

@johanndev answer is correct.
